earthfile2llb/wait_block.go

```diff
@@ -232,16 +232,17 @@ func (wb *waitBlock) waitStates(ctx context.Context) error {
 	sem := semutil.NewMultiSem(sharedParallelism, semutil.NewWeighted(1))
 
 	errGroup, ctx := serrgroup.WithContext(ctx)
-	for _, item := range stateItems {
-		errGroup.Go(func() error {
-			rel, err := sem.Acquire(ctx, 1)
-			if err != nil {
-				return errors.Wrapf(err, "acquiring parallelism semaphore during waitStates for %s", item.c.target.String())
-			}
-			defer rel()
-
-			return item.c.forceExecution(ctx, *item.state, item.c.platr)
-		})
+	for _, itemItr := range stateItems {
+		func(item *stateWaitItem) {
```

Comment:
I think we can also use the `item := item` idiom to avoid the extra func wrapper.

https://riptutorial.com/go/example/3897/using-closures-with-goroutines-in-a-loop

Reply:
I considered that, but thought the extra function wrapper made it more explicit (and unlikely for someone to remove in the future).

Reply:
at the same time, a comment saying `item := item // required due to delayed func` might be sufficient in that case. I'll do that.

Reply:
Cool. I don't have a strong opinion. Both approaches aren't exactly intuitive. The `item := item` one is a little tidier but perhaps less obvious.